### ISSUE TITLE
bug: unconditional return in constructor produces unreachable code after class

### ISSUE THREAD

**Issue by Xernisfy** [denoland/deno_lint#353]
An unconditional return statement inside of a class constructor also leads to a "unreachable" linting error:

```ts
class Class {
  constructor() {
    return this;
  }
}

console.log("unreachable???");
```
```
(no-unreachable) This statement is unreachable
console.log("unreachable???");
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    at \deno-lint.ts:7:0

Found 1 problem
Checked 1 file
```

A conditional return however does not produce an error:
```ts
class Class {
  constructor() {
    if (Deno) return this;
  }
}

console.log("unreachable???");
```

_Originally posted in https://github.com/denoland/deno_lint/issues/318#issuecomment-702154495_

**Comment by magurotuna** [comment]
Thanks for opening an issue!
I've fixed it in #355, so it'll be fixed in the next (or maybe after the next) version of Deno.
